web/src/app/services/temp-survey/temp-survey.service.ts

```diff
@@ -0,0 +1,109 @@
+import {Injectable} from '@angular/core';
+import {Survey} from 'app/models/survey.model';
+import {DataStoreService} from '../data-store/data-store.service';
+import {BehaviorSubject, Observable, firstValueFrom} from 'rxjs';
+import {LocationOfInterest} from 'app/models/loi.model';
+import {List} from 'immutable';
+import {Job} from 'app/models/job.model';
+import { Task } from 'app/models/task/task.model';
+import { LocationOfInterestService } from '../loi/loi.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class TempSurveyService {
+  private tempSurvey$$!: BehaviorSubject<Survey>;
+  private tempLois$$!: BehaviorSubject<List<LocationOfInterest>>;
+
+  private tempSurvey!: Survey;
```

Comment:
I see below we're using `this.tempSurvey$$.getValue()` to access the current survey. In that cases, would this field be redundant?